File: engine/src/main/java/ru/vertuos/engine/world/handlers/gravity/VerticalGravityResolver.java
package ru.vertuos.engine.world.handlers.gravity;

import ru.vertuos.engine.world.object.DynamicGameObject;

public class VerticalGravityResolver
    implements GravityResolver<DynamicGameObject> {

    private static final float G = 15f;

    @Override
    public void resolve(DynamicGameObject object, float dt) {
        if (!object.isOnGround()) {
            object.setAccelerationY(-G);
        }
    }
}

File: engine/src/main/kotlin/ru/vertuos/engine/world/updaters/gravity/VerticalGravityResolver.kt
package ru.vertuos.engine.world.updaters.gravity

import ru.vertuos.engine.util.setLinearVelocityY
import ru.vertuos.engine.world.obj.DynamicGameObject

class VerticalGravityResolver : GravityResolver<DynamicGameObject> {

    override fun resolve(obj: DynamicGameObject, dt: Float) {
        if (!obj.isOnGround) {
            obj.setLinearVelocityY(obj.linearVelocity.y - G * obj.mass * dt)
        }
    }

    companion object {

        private const val G = 15f
    }
}
